WalletWasabi.Fluent/ViewModels/SearchBar/Sources/TransactionsSearchSource.cs

```diff
@@ -117,6 +119,28 @@ private static IEnumerable<ISearchItem> Search(string query)
 	private static IEnumerable<(WalletViewModel, HistoryItemViewModelBase)> Filter(string queryStr)
 	{
 		return Flatten(GetTransactionsByWallet())
-			.Where(tuple => ContainsId(tuple.Item2, queryStr));
+		.Where(tuple => IsBitcoinAddress(tuple.Item1.WalletModel.Network, queryStr) ?
+			ContainsDestinationAddress(tuple, queryStr) :
+			ContainsId(tuple.Item2, queryStr));
+	}
+
+	private static bool ContainsDestinationAddress((WalletViewModel, HistoryItemViewModelBase) tupleWalletHistoryItem, string queryStr)
+	{
+		var txid = tupleWalletHistoryItem.Item2.Transaction.Id;
+		return tupleWalletHistoryItem.Item1.WalletModel.Transactions.GetDestinationAddresses(txid)
```

Comment:
`GetDestinationAddresses`? There is no such a thing as addresses. What about scripts that cannot be encoded as an "address"? Take care of this, it looks weird.

Reply:
> `GetDestinationAddresses`? There is no such a thing as addresses. What about scripts that cannot be encoded as an "address"? Take care of this, it looks weird.

for this part only an address can be passed on the `ContainsDestinationAddress` method. A check `TryParseBitcoinAddress` has already been done before it was called.